src/client/src/containers/SelectOption/index.tsx

```diff
@@ -34,28 +61,30 @@ interface ISelectOptionProps {
 
 interface ISelectOptionState {
   selectedCardIndices: number[];
+  maxPageReached: boolean;
+  description: string;
 }
 
 interface IDispatchProps {
   setDetailPage: (detailPageInfo: IOption) => void;
 }
 
-type Props = IDispatchProps & ISelectOptionProps;
+type Props = IDispatchProps & ISelectOptionProps & IProps;
 
 class SelectOption extends React.Component<Props, ISelectOptionState> {
   constructor(props: Props) {
     super(props);
     const { selectedCardIndices } = props;
     this.state = {
-      selectedCardIndices
+      selectedCardIndices,
+      maxPageReached: false,
+      description: props.intl.formatMessage(messages.limitedPages)
     };
+    this.addPage = this.addPage.bind(this);
```

Comment:
We could avoid this binding by using arrow properties:
https://www.freecodecamp.org/news/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56/

Reply:
That's good to know!